src/pyprojroot/root.py

```diff
@@ -15,13 +15,19 @@
 )
 
 
-def as_start_path(start: Union[None, _PathType]) -> Path:
-    if start is None:
-        return Path.cwd()
-    if not isinstance(start, Path):
-        start = Path(start)
-    # TODO: consider `start = start.resolve()`
-    return start
+def as_start_path(start: Optional[_PathType]) -> Path:
+    """
+    Returns a Path object based on the current working directory or the optional input
+    provided. If the input `start` parameter contains the '~' character, it will be
+    expanded to the home directory before being returned.
+
+    :param start: Optional[str or Path], the path to start from. Defaults to None
+                  which sets the starting path to the current working directory.
+    :return: Path, the Path object based on the starting path.
+    """
+    if start is not None:
+        return Path(start).expanduser()
```

Comment:
(minor) I'm not sure it's worth converting an existing Path object into a new one. But I'm also not sure there's any harm.

Reply:
I have pushed to the change some tests, however, I'm not sure if I've set them up properly.  Currently the existing test imports from the installed version, I changed it to look at the local.  I'm not sure of the expectation here.

Let me know if I'm missing any tests / updates needed.   

Also I did remove the `isinstance` as there wasn't any harm.  Let me know if its still more readable for you if the check is there.

Reply:
Thanks. I'll take a look over the next few days.

Sorry if my comments have been a bit terse---I haven't had the time or energy for more---but I appreciate you trying to help.

Reply:
Totally understand.  I appreciate your time as well.  Will attempt to finalize this tomorrow.